### DISCUSSION TITLE
unknown parameter to use bsimcmg107 model

### DISCUSSION THREAD

**Discussion by selldream2** [pascalkuthe/OpenVAF#115] [Q&A]
Sir,

I try to make OSDI with bsimcmg107 model in ngspice.
compiling is good to make osdi file.  However, I can meet "unknown parameter" error for all parameter.
How can I solve this problem?

![image](https://github.com/pascalkuthe/OpenVAF/assets/27531428/1251abea-0dda-4692-9203-3ad928a6bd0e)

**Comment by dwarning** [comment]
Bsimcmg version 107 is meanwhile a bit outdated. The original model from 2013 make no difference between instance an model parameter in the va code.
In case you want work with this old mode you have to change the parameter description for instance parameter in bsimcmg_body.include e.g.:
parameter real TFIN      =  15n from [1n:inf);      // Body (Fin) thickness
into:
(* type="instance" *)  parameter real TFIN      =  15n from [1n:inf);      // Body (Fin) thickness

or better use the newer versions 110 or 111, e.g. start from here [https://github.com/dwarning/VA-Models](url)

See also #96
